projects/js-packages/publicize-components/src/components/admin-page/test/social-module-toggle.test.jsx

```diff
@@ -0,0 +1,78 @@
+// eslint-disable-next-line import/order -- This is a test file and we need to import the mocks first
+import { mockStore } from '../../../utils/test-mocks';
+import { getScriptData } from '@automattic/jetpack-script-data';
+import { render, screen } from '@testing-library/react';
+import { useDispatch } from '@wordpress/data';
+import { getSocialScriptData, hasSocialPaidFeatures } from '../../../utils';
+import SocialModuleToggle from '../toggles/social-module-toggle';
+
+jest.mock( '../../connection-management', () => () => <div data-testid="connection-management" /> );
+
+describe( 'SocialModuleToggle', () => {
+	const mockUpdateSettings = jest.fn();
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+		mockStore();
+
+		// Mock dispatch to capture updateSocialModuleSettings calls
+		useDispatch.mockReturnValue( {
+			updateSocialModuleSettings: mockUpdateSettings,
+		} );
+
+		// Default script data mocks
+		getScriptData.mockReturnValue( {
+			site: {
+				wpcom: { blog_id: '123' },
+			},
+		} );
+
+		getSocialScriptData.mockReturnValue( {
+			urls: { connectionsManagementPage: 'https://example.com/connections' },
+			feature_flags: { useAdminUiV1: false },
```

Comment:
The default is for this to be true. Do you think we should focus the majority of the tests on this being true so that when we remove the feature flag, there is less to update?

Reply:
Good idea!

Reply:
Done in a61d62d